apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -99,15 +96,14 @@ private KeySet(bool allKeys, IEnumerable<Key> keys, IEnumerable<KeyRange> ranges
             Ranges = ranges;
         }
 
-        /// <inheritdoc/>
-        public override bool Equals(object other)
-            => other is KeySet keySet && Protobuf.Equals(keySet.Protobuf);
-
-        /// <inheritdoc/>
-        public override int GetHashCode() => Protobuf.GetHashCode();
-
-        /// <inheritdoc/>
-        public override string ToString()
+        /// <summary>
+        /// Returns the string representation of this KeySet using the type conversion options specified in 
+        /// <see cref="SpannerConnectionStringBuilder"/>. The string representation is only for diagnostic purposes
+        /// and the implementation may change over time.
+        /// </summary>
+        /// <param name="builder">The <see cref="SpannerConnectionStringBuilder"/> instance.</param>
```

Comment:
I'd expand this description to explain the purpose of the builder - i.e. that it's used to derive default conversion options.